### PR TITLE
add gas estimation

### PR DESCRIPTION
**Pull request by altbee**

### CODE REVIEW

File: liquidator/liquidate.js [hunk 1]
```diff
@@ -87,15 +88,27 @@ exports.liquidate = async (unhealthyAccount, liquidationAmount) => {
...
+      console.log("gas estimation error");
+    }
+    if (!estimatedGas.isZero()) {
+      const liquidationTxObj = await liquidationTx.send({
+        from: myAccount,
+        gasLimit: calculateGasLimit(estimatedGas),
+        gasPrice,
+      });
+
+      // TODO: for now, this account should have enough arUSD manually deposited by human
```
Comment (gophereth) [inline liquidator/liquidate.js:108]: Does it stop liquidation transaction if it does not have arUSD?

File: liquidator/liquidate.js [hunk 2]
```diff
@@ -87,15 +88,27 @@ exports.liquidate = async (unhealthyAccount, liquidationAmount) => {
...
+    try {
+      estimatedGas = await liquidationTx.estimateGas({
+        from: myAccount,
+        gasLimit: 300000,
+        gasPrice,
+      });
+    } catch (error) {
+      console.log("gas estimation error");
+    }
+    if (!estimatedGas.isZero()) {
```
[the remark this answers is not in the document]
Comment (altbee) [reply liquidator/liquidate.js:101]: done

File: liquidator/liquidate.js (hunk 1 above)
Comment (gophereth) [reply liquidator/liquidate.js:108]: Do you think if arUSD does not exist, estimation would fail or not?

File: liquidator/liquidate.js [hunk 3]
```diff
@@ -87,15 +88,27 @@ exports.liquidate = async (unhealthyAccount, liquidationAmount) => {
...
+      const liquidationTxObj = await liquidationTx.send({
+        from: myAccount,
+        gasLimit: calculateGasLimit(estimatedGas),
+        gasPrice,
+      });
+
+      // TODO: for now, this account should have enough arUSD manually deposited by human
+      // TODO: implement the mechanism to convert liquidated funds into arUSD or into USDT - the point when to do it - do it instantly?
```
Comment (gophereth) [inline liquidator/liquidate.js:109]: Will be done by human

**Pull request merged**